src/aap_eda/core/tasking/__init__.py

```diff
@@ -204,14 +207,15 @@ def unique_enqueue(queue_name: str, job_id: str, *args, **kwargs) -> Job:
     Detects if a job with the same id is already enqueued and if it is
     it will return it instead of enqueuing a new one.
     """
-    queue = get_queue(queue_name)
-    job = job_from_queue(queue, job_id)
-    if job:
-        logger.info(
-            f"Skip enqueing job: {job_id} because it is already enqueued"
-        )
-        return job
+    for name in settings.RQ_QUEUES:
+        job = job_from_queue(name, job_id)
+        if job:
+            logger.info(
+                f"Skip enqueing job: {job_id} because it is already enqueued"
+            )
+            return job
     else:
+        queue = get_queue(name=queue_name)
```

Comment:
Doesn't this mean that a job could be sent to a different queue than the one on which it was previously run?  Perhaps only used when there is an issue with the job's original queue?  Could be that the queuing manipulation isn't fully implemented yet?  If that's the case a comment would be good.

Reply:
The monitor logic will schedule a job in default queue before get enqueued in the right activation queue. The uniqueness must be guaranteed for all the queues. Also there are cases where an activation might change the queue if it is restarted manually and the node is unhealthy.